neo/Ledger/MemoryPool.cs

```diff
@@ -360,8 +360,10 @@ internal void UpdatePoolForBlockPersisted(Block block, Snapshot snapshot)
 
                 if (policyChanged)
                 {
+                    long newFeePerByte = NativeContract.Policy.GetFeePerByte(snapshot);
                     foreach (PoolItem item in _unverifiedSortedTransactions.Reverse())
-                        _system.Blockchain.Tell(item.Tx, ActorRefs.NoSender);
+                        if(item.Tx.FeePerByte <= newFeePerByte)
```

Comment:
No. The real `feePerByte` must subtract the fee consumed by `Verification`.

Reply:
I did not understand, sorry, @erikzhang.

`newFeePerByte` would be the new Policy, right?
`item.Tx.FeePerByte` is the current fee that the `tx` used for paying its fees.

Just we need to switch the check as @lock9 suggested?

Reply:
@erikzhang, please check it now, formula has been reverted as previously suggested by @lock9.

Reply:
https://github.com/neo-project/neo/blob/54e772ae594aeea9ca1a774a79dc2d0f2fd361c9/neo/Network/P2P/Payloads/Transaction.cs#L190-L192

Your code is not compatible with this. I think it's impossible to optimize.

Reply:
I am not understanding, @erikzhang. Please, check my line of reasoning and sorry for the mistakes:

1. the `tx_1` is in the mempool has has `FeePerByte = 1000`.
https://github.com/neo-project/neo/blob/54e772ae594aeea9ca1a774a79dc2d0f2fd361c9/neo/Network/P2P/Payloads/Transaction.cs#L45

It is a fixed public long static determined by the `NetworkFee` and `Size`.

2. A block is generated and `tx_1` is not included
3. Policy changes and min `FeePerByte` is increased from `800` to `1000`.
4. `tx_1` is still valid because 1000 >= 1000, otherwise, it should not be re-verified.

Reply:
`Verification` script may has consumed some gas.

Reply:
Exactly, Erik, we still need to check this, as well as the GAS possibly spent during `Application`.

But this is a filter, Erik, in the worse case we will still call `_system.Blockchain.Tell(item.Tx, ActorRefs.NoSender);`. In the good cases we are discarding those that are in fact invalid.